### DISCUSSION TITLE
Memory Leak with uvicorn = "^0.24.0"

### DISCUSSION THREAD

**Discussion by saka-bang** [Kludex/uvicorn#2194] [Q&A]
I am observing memory leak issues with "^0.24.0"
however when I downgrade to "^0.22.0" memory leak disappear.
I have tested with same source, infra and configuration. Issue is reproducible by just upgrading uvicorn version

[1] below with "^0.24.0"
<img width="353" alt="image" src="https://github.com/encode/uvicorn/assets/133739101/277cee2b-2dac-49dd-b091-41b5c8728a8f">

[2] below with "^0.22.0"
<img width="353" alt="image" src="https://github.com/encode/uvicorn/assets/133739101/db12cc92-f1ba-4deb-bd64-6672211deefc">

**Comment by saka-bang** [comment]
I will however go through this discussion again and add additional details. It is not the similar case but we do have FastAPI app running on this instance.

https://github.com/encode/uvicorn/discussions/1994

**Comment by Kludex** [comment]
How about 0.23.0?

**Reply by saka-bang** [reply]
Good question. I will try with 0.23.0 and share results soon

**Comment by saka-bang** [comment]
we are using uvicorn workers with gunicorn

<img width="355" alt="image" src="https://github.com/encode/uvicorn/assets/133739101/d7af4c71-bf3b-43a6-b6d9-c136af619559">

**Reply by Kludex** [reply]
What if you don't?

**Reply by saka-bang** [reply]
It seems with default worker type it is not producible, I will try couple of more options and share report

**Comment by trisduong** [comment]
I met this problem too. Any update on that?. Currently, We will try with the newer version 0.25.0

**Reply by felippem** [reply]
Searching, I found this related issue. https://github.com/python/cpython/issues/109534

**Reply by trisduong** [reply]
Currently, we are not using uvloop and tls/ssl as well but the issue still occurs :((
